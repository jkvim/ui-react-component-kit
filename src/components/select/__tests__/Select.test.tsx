import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Select } from '../';

describe('Select', () => {

  const option = { display: 'Test Display', value: 'Test Value'};

  describe('props', () => {

    const mockCallback = jest.fn();

    const wrapper = mount(
      <Select
        id="id"
        label="label"
        placeholder="placeholder"
        options={[option]}
        value="Test Value"
        hint="hint"
        disabled={true}
        errorMessage="message"
        className="class"
        onBlur={mockCallback}
        onChange={mockCallback}
      />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `id` in props', () => {
      expect(wrapper.props().id).toBe('id');
    });

    it('should pass `label` in props', () => {
      expect(wrapper.props().label).toBe('label');
    });

    it('should pass `placeholder` in props', () => {
      expect(wrapper.props().placeholder).toBe('placeholder');
    });

    it('should pass `options` in props', () => {
      expect(wrapper.props().options).toEqual([option]);
    });

    it('should pass `value` in props', () => {
      expect(wrapper.props().value).toBe('Test Value');
    });

    it('should pass `hint` in props', () => {
      expect(wrapper.props().hint).toBe('hint');
    });

    it('should pass `disabled` in props', () => {
      expect(wrapper.props().disabled).toBe(true);
    });

    it('should pass `errorMessage` in props', () => {
      expect(wrapper.props().errorMessage).toBe('message');
    });

    it('should pass `className` in props', () => {
      expect(wrapper.props().className).toBe('class');
    });

    it('should pass `onBlur` in props', () => {
      expect(wrapper.props().onBlur).toBe(mockCallback);
    });

    it('should pass `onChange` in props', () => {
      expect(wrapper.props().onChange).toBe(mockCallback);
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(<Select id="id" label="label" options={[option]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot error', () => {
    const wrapper = mount(<Select id="id" label="label" errorMessage="message" options={[option]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot disabled', () => {
    const wrapper = mount(<Select id="id" label="label" disabled={true} options={[option]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('expect value to change when keys are pressed', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<Select id="id" label="label" onChange={mockCallback} options={[option]} />);
    wrapper.find('select').simulate('change', { target: { value:  'Test Value'}});
    // The mock function is called once
    expect(mockCallback).toBeCalled();
    // The new value was sent
    expect(mockCallback.mock.calls[0][0]).toBe('Test Value');
  });

  it('expect everything to work when no onChange is provided', () => {
    const wrapper = mount(<Select id="id" label="label" options={[option]}  />);
    wrapper.find('select').simulate('change', { target: { value: 'Test Value' } });
  });

  it('expect blur to occur when focused away', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<Select id="id" label="label" options={[option]} onBlur={mockCallback} />);
    wrapper.find('select').simulate('blur');
    // The mock function is called once
    expect(mockCallback).toBeCalled();
  });

  it('expect everything to work when no onBlur is provided', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<Select id="id" label="label" options={[option]} onBlur={mockCallback} />);
    wrapper.find('select').simulate('blur');
  });
});
