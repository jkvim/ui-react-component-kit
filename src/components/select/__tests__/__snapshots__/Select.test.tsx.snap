// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField props should match snapshot 1`] = `
<Select
  id="some-id"
  label="Some error message."
  options={
    Array [
      Object {
        "display": "Test Display",
        "value": "Test Value",
      },
    ]
  }
  placeholder="Please select"
>
  <styled.div>
    <div
      className="sc-gzVnrw kDUAmj"
    >
      <styled.label
        hasError={false}
        htmlFor="some-id"
        theme={
          Object {
            "color": Object {
              "color": "#333",
              "errorColor": "#9f173f",
            },
          }
        }
      >
        <label
          className="sc-bdVaJa dWyWPR"
          htmlFor="some-id"
        >
          Some error message.
        </label>
      </styled.label>
      <styled.select
        hasError={false}
        id="some-id"
        innerRef={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        theme={
          Object {
            "color": Object {
              "borderColor": "#9e9e9e",
              "errorColor": "#9f173f",
            },
          }
        }
        value=""
      >
        <select
          className="sc-bZQynM gHiYcJ"
          id="some-id"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Please select
          </option>
          <option
            key="0"
            value="Test Value"
          >
            Test Display
          </option>
        </select>
      </styled.select>
      <styled.div
        theme={
          Object {
            "color": Object {
              "errorColor": "red",
            },
          }
        }
      >
        <div
          className="sc-gqjmRU gsDxgu"
        />
      </styled.div>
    </div>
  </styled.div>
</Select>
`;
